src/query/sql/src/planner/optimizer/rule/rewrite/rule_fold_count_aggregate.rs

```diff
@@ -101,8 +112,36 @@ impl Rule for RuleFoldCountAggregate {
                 eval_scalar.into(),
                 SExpr::create_leaf(dummy_table_scan.into()),
             ));
+        } else if let (true, column_stats, Some(table_card)) = (
+            simple_nullable_count,
+            input_prop.column_stats,
+            input_prop.precise_cardinality,
+        ) {
+            let mut scalars = agg.aggregate_functions;
+            // Now count function must only have one arg.
+            for item in scalars.iter_mut() {
+                if let Scalar::AggregateFunction(agg_func) = item.scalar.clone() {
+                    let col_set = agg_func.args[0].used_columns();
+                    for index in col_set {
+                        let col_stat = column_stats.get(&index);
```

Comment:
The index maybe not true column_stats needs index.

```rust
CREATE TABLE `t` (
  `id` INT,
  `b` TUPLE(INT, INT),
  `c` VARCHAR NULL
)

mysql> insert into t(id) values(0),(1),(2),(3);
Query OK, 0 rows affected (0.03 sec)

mysql> select * from t;
+------+-------+------+
| id   | b     | c    |
+------+-------+------+
|    0 | (0,0) | NULL |
|    1 | (0,0) | NULL |
|    2 | (0,0) | NULL |
|    3 | (0,0) | NULL |
+------+-------+------+
4 rows in set (0.04 sec)
Read 4 rows, 89.00 B in 0.020 sec., 198.4 rows/sec., 4.31 KiB/sec.

mysql> select count(c) from t;
+----------+
| count(c) |
+----------+
|        4 |
+----------+
1 row in set (0.02 sec)
Read 1 rows, 1.00 B in 0.002 sec., 461.1 rows/sec., 461.10 B/sec.
```

In BoundColumn the index of col b is 1, index of col c is 2.

But in RelationalProperty the index of col c is 3.

So, I convert this pr state to a draft.